### PR TITLE
Documentation: Add extra terminology to the key concepts glossary

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Tweak some of the key concepts in the documentation.

## Additional info

Please review whether the terminology is right or should be improved.
This tries to resolve some of the issues with [undocumented terminology I mentioned here](https://community.ynput.io/t/subset-and-family-are-they-the-right-names/104/14?u=bigroy).

## Testing notes:

1. Read the glossary

### CODE REVIEW

File: website/docs/artist_concepts.md [hunk 1]
```diff
@@ -14,13 +14,25 @@ OpenPype has a limitation regarding duplicated names. Name of assets must be uni
 
 ### Subset
 
-Usually, an asset needs to be created in multiple *'flavours'*. A character might have multiple different looks, model needs to be published in different resolutions, a standard animation rig might not be usable in a crowd system and so on. 'Subsets' are here to accommodate all this variety that might be needed within a single asset. A model might have subset: *'main'*, *'proxy'*, *'sculpt'*, while data of *'look'* family could have subsets *'main'*, *'dirty'*, *'damaged'*. Subsets have some recommendations for their names, but ultimately it's up to the artist to use them for separation of publishes when needed.
+A published output from an asset results in a subset
```
Comment (tokejepsen) [inline website/docs/artist_concepts.md:17]:
```suggestion
A published output from an asset results in a subset.
```

File: website/docs/artist_concepts.md [hunk 2]
```diff
@@ -14,13 +14,25 @@ OpenPype has a limitation regarding duplicated names. Name of assets must be uni
...
+
+The subset type is referred to as [family](#family), for example a rig, pointcache, look.
+A single asset can have many subsets, even of a single family, named [variants](#variant).
+By default a subset is named as a combination of family + variant, sometimes prefixed with the task name (like workfile)
```
Comment (tokejepsen) [inline website/docs/artist_concepts.md:21]:
```suggestion
By default a subset is named as a combination of family + variant, sometimes prefixed with the task name (like workfile).
```

File: website/docs/artist_concepts.md [hunk 3]
```diff
@@ -33,18 +45,22 @@ At this moment names of assets, tasks, subsets or representations can contain on
 
 ### Family
 
-Each published [subset][3b89d8e0] can have exactly one family assigned to it. Family determines the type of data that the subset holds. Family doesn't dictate the file type, but can enforce certain technical specifications. For example OpenPype default configuration expects `model` family to only contain geometry without any shaders or joints when it is published.
+Each published [subset][#subset] can have exactly one family assigned to it. Family determines the type of data that the subset holds. Family doesn't dictate the file type, but can enforce certain technical specifications. For example OpenPype default configuration expects `model` family to only contain geometry without any shaders or joints when it is published.
 
+### Task
 
-  [3b89d8e0]: #subset "subset"
+A task defines a work area for an asset an artist can work in. For example asset *hero* can have tasks *modeling* and *rigging*. Tasks also have types. Multiple tasks of the same type may exist in an asset. A task with type FX could for example appear twice as *fx_fire* and *fx_cloth*.
```
Comment (tokejepsen) [inline website/docs/artist_concepts.md:52]:
```suggestion
A task defines a work area for an asset where an artist can work in. For example asset *characterA* can have tasks named *modeling* and *rigging*. Tasks also have types. Multiple tasks of the same type may exist on an asset. A task with type `fx` could for example appear twice as *fx_fire* and *fx_cloth*.
```

"Hero" might be confused with hero version.

File: website/docs/artist_concepts.md [hunk 4]
```diff
@@ -54,6 +70,10 @@ Small piece of software usually dedicated to a particular purpose. Most of OpenP
 
 Process of exporting data from your work scene to versioned, immutable file that can be used by other artists in the studio.
 
+#### (Publish) Instance
+
+A publish instance is a single entry which defines a publish output. Publish instances persist with the workfile. This way we can expect that a publish from a newer workfile will produce similar consistent versioned outputs.
```
Comment (tokejepsen) [inline website/docs/artist_concepts.md:75]:
```suggestion
A publish instance is a single entry which defines a publish output. Publish instances persist within the workfile. This way we can expect that a publish from a newer workfile will produce similar consistent versioned outputs.
```

**Review by tokejepsen — CHANGES_REQUESTED** [review]
Looks good! Minor changes.

**Review by tokejepsen — APPROVED** [review]

**Review by antirotor — APPROVED** [review]

**Pull request merged**